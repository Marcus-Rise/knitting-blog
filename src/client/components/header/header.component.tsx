import type { FC } from "react";
import React from "react";
import type { INavLink } from "../nav";
import { Nav } from "../nav";
import Link from "next/link";
import NextImage from "next/image";
import { Container, Image, LinkStyled, Root, Title } from "./header.styles";
import Logo from "../../../../public/android-chrome-192x192.png";

interface IProps {
  title: string;
  links: ReadonlyArray<INavLink>;
}

const LOGO_SIZE = 35;

const Header: FC<IProps> = ({ links, title }) => (
  <Root>
    <Link href={"/"}>
      <LinkStyled>
        <Container>
          <Title>{title}</Title>
          <Image
            as={NextImage}
            src={Logo}
            alt={title}
            height={LOGO_SIZE}
            width={LOGO_SIZE}
            quality={15}
          />
        </Container>
      </LinkStyled>
    </Link>
    <Nav items={links} />
  </Root>
);

export { Header };
