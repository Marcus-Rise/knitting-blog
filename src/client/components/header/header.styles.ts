import styled from "styled-components";
import { BadScript, device } from "../../styles";

const Root = styled.header`
  display: flex;
  flex-direction: column;

  @media (${device.md}) {
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    position: sticky;
    top: 0;
    z-index: 100;
    background: white;
    padding: 1rem;
    box-shadow: 0 5px 5px 0 rgba(50, 50, 50, 0.25);
  }

  @media (${device.sm}) {
    padding: 0.5rem;
  }
`;

const LinkStyled = styled.a`
  color: inherit;
  text-decoration: none;
`;

const Container = styled.div`
  padding-top: 1.75rem;
  padding-bottom: 1.75rem;
  box-sizing: border-box;

  background-image: url("../../assets/header_background_left.png"),
    url("../../assets/header_background_right.png");
  background-repeat: no-repeat, no-repeat;
  background-size: contain, contain;
  background-position: left top, right bottom;

  @media (${device.md}) {
    background: none;
    display: flex;
    align-items: center;
    padding: 0;
  }
`;

const Title = styled.h1`
  ${BadScript};
  font-size: 2.5rem;
  text-align: center;

  @media (${device.md}) {
    margin: 0;
    font-size: 1.5rem;
  }
`;

export { Root, Title, LinkStyled, Container };
