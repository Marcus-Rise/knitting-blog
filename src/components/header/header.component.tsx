import type { FC } from "react";
import styles from "./header.module.scss";
import Link from "next/link";
import Image from "next/image";
import { KnitsLeftIcon, KnitsRightIcon } from "../icons";
import classNames from "classnames";
import logo from "../../app/icon.png";
import { Theme } from "../theme";

const LOGO_SIZE = 35;

type Props = { title: string };

const Header: FC<Props> = ({ title }) => (
  <header className={styles.header}>
    <KnitsLeftIcon
      className={classNames(styles.knits, styles.knitsLeft)}
      width={"10rem"}
      height={"6rem"}
    />
    <KnitsRightIcon
      className={classNames(styles.knits, styles.knitsRight)}
      width={"9rem"}
      height={"10rem"}
    />
    <Link className={styles.link} href={"/"}>
      <div className={styles.logo}>
        <h1 className={styles.title}>{title}</h1>
        <Image
          alt={title}
          src={logo}
          height={LOGO_SIZE}
          width={LOGO_SIZE}
          quality={15}
          className={styles.image}
        />
      </div>
    </Link>
    <Theme className={styles.theme} />
  </header>
);

export { Header };
