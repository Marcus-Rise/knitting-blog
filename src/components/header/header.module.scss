@use "../../styles/media" as m;
@use "../../styles/typography" as t;

.header {
  display: flex;
  flex-direction: column;

  @include m.md {
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    position: sticky;
    top: 0;
    z-index: 100;
    background: white;
    padding: 1rem;
    box-shadow: 0 5px 5px 0 rgb(50 50 50 / 25%);
  }

  @include m.md {
    padding: 0.5rem 1rem;
  }
}

.logo {
  padding-top: 1.75rem;
  padding-bottom: 1.75rem;
  box-sizing: border-box;
  background-image: url("/header_background_left.png"), url("/header_background_right.png");
  background-repeat: no-repeat, no-repeat;
  background-size: contain, contain;
  background-position: left top, right bottom;

  @include m.md {
    background: none;
    display: flex;
    align-items: center;
    align-content: center;
    padding: 0;
    gap: 1rem;
  }
}

.image {
  @include m.lg {
    display: none !important;
  }
}

.link {
  @include t.bad-script;

  color: inherit;
  text-decoration: none;
  font-size: 2.5rem;
  text-align: center;

  &:hover {
    cursor: pointer;
  }

  @include m.md {
    margin: 0;
    font-size: 1.5rem;
  }
}
