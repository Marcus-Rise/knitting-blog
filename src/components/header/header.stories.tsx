import type { ComponentMeta, ComponentStory } from "@storybook/react";
import { Header } from "./header.component";

const Config: ComponentMeta<typeof Header> = {
  title: "components/Header",
  component: Header,
};

const Template: ComponentStory<typeof Header> = (args) => <Header {...args} />;

const Default = Template.bind({});
Default.args = {
  title: "Надя вяжет",
};

export default Config;
export { Default };
