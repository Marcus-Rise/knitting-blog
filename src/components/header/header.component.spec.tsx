import { render } from "@testing-library/react";
import { Header } from "./header.component";
import React from "react";

jest.mock("../nav", () => ({
  __esModule: true,
  Nav: jest.fn(() => <nav />),
}));
jest.mock("next/image", () => ({
  __esModule: true,
  default: jest.fn((props) => (
    <img src={props.logoSrc} alt={props.title} height={props.logoSize} width={props.logoSize} />
  )),
}));

describe("Header", () => {
  test("render", () => {
    const { asFragment } = render(
      <Header title={"title"} logoSrc={"http://test.com"} links={[{ title: "RRR", link: "#" }]} />,
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
