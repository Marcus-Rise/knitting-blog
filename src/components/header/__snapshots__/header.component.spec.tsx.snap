// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header render 1`] = `
<DocumentFragment>
  <header
    class="root"
  >
    <a
      class="link"
    >
      <div
        class="logoContainer"
      >
        <div
          class="logo"
        >
          <img />
        </div>
        <h1
          class="title"
        >
          title
        </h1>
      </div>
    </a>
    <nav />
  </header>
</DocumentFragment>
`;
